src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -61,8 +61,9 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                        UiRequiredExceptionClassification.AcquireTokenSilentFailed);
                 }
 
-                bool isAccountSourceDeviceCodeFlow = !string.IsNullOrEmpty(AuthenticationRequestParameters.Account.AccountSource) &&
-                                               AuthenticationRequestParameters.Account.AccountSource == "device_code_flow";
+                var account = AuthenticationRequestParameters.Account as Account;
+                bool isAccountSourceDeviceCodeFlow = !string.IsNullOrEmpty(account.AccountSource) &&

```

Comment:
There can be a null pointer ex here if account is null. 

Try smth like `bool isAccountSourceDeviceCodeFlow = account != null && account.AccountSource == "device_code_flow"`